enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,613 @@
+3. The cluster creator obtains the OpenShift installer in one of [the
+   usual
+   ways](https://docs.openshift.com/container-platform/4.9/installing/installing_bare_metal/installing-bare-metal.html#installation-obtaining-installer_installing-bare-metal).
+4. The cluster creator triggers their orchestration tool to run the
+   installer.
+5. The orchestration tool prepares an `install-config.yaml`, including
+   `NMState` content to configure the static network settings for the
+   hosts, using input data and templates.
+6. The orchestration tool runs `openshift-install create image`.
+7. The installer extracts a copy of the RHCOS ISO from the OpenShift
+   release payload.
+8. The installer combines the inputs provided by the user and produces
+   an Ignition config.
+9. The installer combines the Ignition config and the RHCOS ISO to
+   create the bootable image configured to run the assisted service and/or
+   agent based on the host that boots the image and writes the new
+   image and cluster credentials to disk.
+10. The orchestration tool copies the ISO to a location that the hosts
+    or VMs can boot from it. For bare metal, that will generally be a
+    web server visible to the management controllers in the bare metal
+    hosts (because the management controllers are unlikely to be on
+    the same network as the external interfaces for the cluster
+    hosts). For vSphere or other hypervisors, that may be a filesystem
+    on the hypervisor host.
+11. The orchestration tool configures the host to boot the ISO. For
+    bare metal, that means configuring the baseboard management
+    controller (BMC) in each host to expose the ISO through the
+    virtual media interface. For vSphere or other hypervisors, similar
+    settings of the VM need to be adjusted.
+12. The orchestration tool boots the hosts.
+13. The orchestration tool uses `openshift-install agent wait-for` to
+    watch the deployment progress and wait for it to complete.
+14. On each host, the startup script in the image selects the correct
+    network settings based on the MAC addresses visible and applies
+    the network settings for each interface.
+15. On node0, the startup script in the image recognizes that it is on
+    node0 and launches the assisted service and the create-cluster
+    service to drive the deployment.
+16. On node0, the create-cluster service waits for the assisted
+    service to start, then creates an InfraEnv using the API.
+17. In parallel
+    - On node0, the create-cluster service waits for the number of
+      agents registered with the service to match the expected value
+      based on the input from the user.
+    - On all nodes (including node0), the startup script in the image
+      connects to the assisted service on node0 to find the InfraEnv
+      ID for the cluster being deployed (there will only be 1). After
```

Comment:
We said that the InfraEnv ID will be baked into the ISO and also passed to the assisted-service, because once auth is enabled agents won't be able to list InfraEnvs.

Reply:
Yep, thanks, I missed updating this step.

Reply:
I updated this step and made a few related changes in 6ce0fe66